### PR TITLE
Fix issue #27

### PR DESCRIPTION
**Pull request by CollinDai**
Created dagger module to supply testing mock data and fix the issue #27

### CODE REVIEW

File: app/src/main/java/ui/com/lightrpg/lightrpgui/game/FakeDataModule.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+
+@Module
+public class FakeDataModule {
+    @Singleton
+    @Provides
+    IDownloadedGameListLoader providesDownloadGameListLoader() {
```
Comment (stevenh47) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/game/FakeDataModule.java]: GameListModule.class is providing this, I don't think we need this one, right?

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/game/FakeDataModule.java]: My mistake. Didn't see that. Let me update

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/game/FakeDataModule.java]: Done

File: app/src/main/java/ui/com/lightrpg/lightrpgui/network/BmobModule.java [hunk 2]
```diff
@@ -24,7 +22,7 @@ IUserService providesUserService() {
 
     @Provides
     @Singleton
-    IGameListLoader providesStoreGameListLoader() {
+    IGameListLoader providesStoreGameListLoader(IDownloadedGameListLoader downloadedGameListLoader) {
```
Comment (stevenh47) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/network/BmobModule.java]: So this is the correct usage and can't be injected to class field?

Comment (CollinDai) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/network/BmobModule.java]: yeah if you want to inject to field you will need to call `injector.inject(this)`. Usually in a module, if you need to build a object with some other object, passing the parameter is good enough. Dagger will go and look for the provider of that method parameter.

Comment (stevenh47) [inline app/src/main/java/ui/com/lightrpg/lightrpgui/network/BmobModule.java]: Oh yes, I forgot to call inject this in this module.

**Pull request merged**